### PR TITLE
Feat/218/allow non Wizeline users to request join a project

### PR DESCRIPTION
**Pull request by AxelUZ**
# What does this PR do?
This "PR" allows users to apply to projects on the "IntershipProjects" page, once they apply the information is displayed in "applicants".

# Where should the reviewer start?
app/routes/internshipProjects/$projectId/index.tsx
app/routes/internshipProjects/$projectId/appliedproject.tsx
app/models/applicant.server.ts
app/routes/applicants/$applicantId/index.tsx
app/routes/projects/$projectId/index.tsx

# How should this be manually tested?
1. Log in through Linkdin, fill in the applicant form and you will be redirected to the "intershipProjects" page.
2.Select any of the displayed projects
3.Click on the "apply" button and you will be redirected to the "intershipProjects" page. (You can repeat this process with the other projects).
4.Verify that if you enter a project to which you have already applied the "apply" button is disabled.
5.Login to your wizeline account, and go to "/applicants".
6.Select the applicant and verify that in the "Applied projects" section appear the projects to which he/she applied. You can also add the "Applied projects" column in "/applicants" to see the projects to which you have applied. 
7.With your Wizeline account, verify that when you enter a project, in the "applicants" section, the applicants that apply to that project appear.

# Any background context you want to provide?

# What are the relevant tickets?
#218  

# Screenshots
Before applying: 
Captura de pantalla 2023-09-29 a la(s) 2 59 56 p m

After applying:
Captura de pantalla 2023-09-29 a la(s) 2 59 49 p m

Applicant column:
Captura de pantalla 2023-09-29 a la(s) 2 58 13 p m

Show detailed information of applied projects:
Captura de pantalla 2023-10-04 a la(s) 12 38 37 p m

Show applicants in projectid page:
Captura de pantalla 2023-10-04 a la(s) 12 38 50 p m

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

**Review by CarlosMilano — COMMENTED** [review]
LGTM!

File: prisma/schema.prisma [hunk 1]
```diff
@@ -437,6 +437,7 @@ model Applicant {
   status                 String    @default("DRAFT")
   mentorId               String?
   comments               String?
+  appliedProjects        String?
```
Comment (martinrobled0) [inline prisma/schema.prisma:440]: Shouldn't we add a relationship to the projects table?

**Review by martinrobled0 — COMMENTED** [review]
One question, what happens if I want to apply for several projects?

**Comment by jackbravo** [comment]
Also, how having the applications visible in the applicant table and the applicant details page is ok. But I'm thinking the most important people that need to know that someone is applying are the members of the team themselves.

How should we notify them or let them know that someone is applying to become a intern/mentee of their project?

**Comment by AxelUZ** [comment]
> One question, what happens if I want to apply for several projects?

You can apply to as many projects as you like, the button will be disabled for projects to which you have already applied.

**Comment by martinrobled0** [comment]
> > One question, what happens if I want to apply for several projects?
> 
> You can apply to as many projects as you like, the button will be disabled for projects to which you have already applied.

ok, I asked because the field is a string and I think that should be an array

**Comment by AxelUZ** [comment]
> > > One question, what happens if I want to apply for several projects?
> > 
> > 
> > You can apply to as many projects as you like, the button will be disabled for projects to which you have already applied.
> 
> ok, nvm I think that we just need to add the relation with the projects table

You're right, I'll modify that too

> Also, how having the applications visible in the applicant table and the applicant details page is ok. But I'm thinking the most important people that need to know that someone is applying are the members of the team themselves.
> 
> How should we notify them or let them know that someone is applying to become a intern/mentee of their project?

Maybe I can add an "applicants" section in the project details. So that when you enter your project you can see who is applying. What do you think?

**Comment by AxelUZ** [comment]
> Also, how having the applications visible in the applicant table and the applicant details page is ok. But I'm thinking the most important people that need to know that someone is applying are the members of the team themselves.
> 
> How should we notify them or let them know that someone is applying to become a intern/mentee of their project?

Maybe I can add an "applicants" section in the project details. So that when you enter your project you can see who is applying. What do you think?

**Comment by martinrobled0** [comment]
> > Also, how having the applications visible in the applicant table and the applicant details page is ok. But I'm thinking the most important people that need to know that someone is applying are the members of the team themselves.
> > How should we notify them or let them know that someone is applying to become a intern/mentee of their project?
> 
> Maybe I can add an "applicants" section in the project details. So that when you enter your project you can see who is applying. What do you think?

Do the mockup and show it us in the channel 😃

File: app/routes/applicants/$applicantId/index.tsx [hunk 2]
```diff
@@ -117,6 +117,21 @@ export default function Applicant() {
     }
   }, [navigation]);
 
+const appliedIdProjects = applicant.appliedProjectsId?.split(',');
+const appliedNameProjects = applicant.appliedProjects?.split(',');
```
Comment (CarlosMilano) [inline app/routes/applicants/$applicantId/index.tsx:121]: why did you use split?

Comment (AxelUZ) [reply app/routes/applicants/$applicantId/index.tsx:121]: To divide the string by "," and convert them into elements of an array. and then save it in "projectMap".

**Review by martinrobled0 — APPROVED** [review]
good job

**Pull request merged**